src/screens/ProfileScreen/index.js

```diff
@@ -419,37 +420,50 @@ const ProfileScreen = ({route}) => {
     });
   };
 
-  const handleUpdateImage = (value, type) => {
-    if (type === 'gallery') {
-      setIsLoadingUpdateImageGalery(true);
-    } else {
-      setIsLoadingUpdateImageCamera(true);
-    }
-    const data = {
-      profile_pic_path: value
-    };
+  const handleUpdateImage = async (imgBase64, type) => {
+    try {
+      if (type === TYPE_GALLERY) {
+        setIsLoadingUpdateImageGallery(true);
+      } else {
+        setIsLoadingUpdateImageCamera(true);
+      }
+      const compressionResult = await ImageCompressionUtils.compress(imgBase64, 'base64');
+      const data = {
+        profile_pic_path: compressionResult
+      };
 
-    updateImageProfile(data)
-      .then(async (res) => {
-        if (type === 'gallery') {
-          setIsLoadingUpdateImageGalery(false);
-        } else {
-          setIsLoadingUpdateImageCamera(false);
-        }
-        if (res.code === 200) {
-          closeImageBs();
-          getMyFeeds();
-          const profilePicture = await fetchMyProfile(true);
-          updateUserClient(profilePicture);
-        }
-      })
-      .catch(() => {
-        if (type === 'gallery') {
-          setIsLoadingUpdateImageGalery(false);
-        } else {
-          setIsLoadingUpdateImageCamera(false);
-        }
+      updateImageProfile(data)
+        .then(async (res) => {
+          if (type === TYPE_GALLERY) {
+            setIsLoadingUpdateImageGallery(false);
+          } else {
+            setIsLoadingUpdateImageCamera(false);
+          }
+          if (res.code === 200) {
+            closeImageBs();
+            getMyFeeds();
+            const profilePicture = await fetchMyProfile(true);
+            updateUserClient(profilePicture);
+          }
+        })
+        .catch(() => {
+          if (type === TYPE_GALLERY) {
+            setIsLoadingUpdateImageGallery(false);
+          } else {
+            setIsLoadingUpdateImageCamera(false);
+          }
+        });
+    } catch (error) {
+      showMessage({
+        message: 'Failed to update profile',
+        type: 'danger'
       });
+      if (type === TYPE_GALLERY) {
+        setIsLoadingUpdateImageGallery(false);
+      } else {
+        setIsLoadingUpdateImageCamera(false);
+      }
+    }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The error handling in this function could be improved. Currently, if an error occurs during the image compression or profile update process, the loading state is not reset, which could leave the user stuck on a loading screen. To fix this, you can add a finally block to ensure that the loading state is always reset, regardless of whether an error occurred.
```diff
    try {
      if (type === TYPE_GALLERY) {
        setIsLoadingUpdateImageGallery(true);
      } else {
        setIsLoadingUpdateImageCamera(true);
      }
      const compressionResult = await ImageCompressionUtils.compress(imgBase64, 'base64');
      const data = {
        profile_pic_path: compressionResult
      };
  
      await updateImageProfile(data)
        .then(async (res) => {
          if (res.code === 200) {
            closeImageBs();
            getMyFeeds();
            const profilePicture = await fetchMyProfile(true);
            updateUserClient(profilePicture);
          }
        });
    } catch (error) {
      showMessage({
        message: 'Failed to update profile',
        type: 'danger'
      });
    } finally {
      if (type === TYPE_GALLERY) {
        setIsLoadingUpdateImageGallery(false);
      } else {
        setIsLoadingUpdateImageCamera(false);
      }
    }
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->